lensless/recon/trainable_recon.py

```diff
@@ -58,29 +65,56 @@ def __init__(self, psf, dtype=None, n_iter=1, **kwargs):
                 Data type to use for optimization.
             n_iter : int
                 Number of iterations for unrolled algorithm.
+            post_process : :py:class:`function` or :py:class:`~torch.nn.Module`, optional
+                If :py:class:`function` : Function to apply to the image estimate after the whole ADMM algorithm. Its input most be (image to process, noise_level), where noise_level is a learnable parameter. If it include aditional learnable parameters, they will not be added to the parameter list of the algorithm. To allow for traning, the function must be autograd compatible.
+                If :py:class:`~torch.nn.Module` : A DruNet like network. The network will be included as a submodule of the algorithm and its parameters will be added to the parameter list of the algorithm. If this isn't intended behavior, set requires_grad=False.
         """
         assert isinstance(psf, torch.Tensor), "PSF must be a torch.Tensor"
         super(TrainableReconstructionAlgorithm, self).__init__(
             psf, dtype=dtype, n_iter=n_iter, **kwargs
         )
+        if isinstance(post_process, torch.nn.Module):
+            # If the post_process is a torch module, we assume it is a DruNet like network.
+            from lensless.utils.image import process_with_DruNet
+
+            self.post_process_model = post_process
+            self.post_process = process_with_DruNet(
+                self.post_process_model, self._psf.device, mode="train"
+            )
+        else:
+            # Otherwise, we assume it is a function.
+            self.post_process = post_process
+        if self.post_process is not None:
+            self.noise_level = torch.nn.Parameter(torch.tensor([1.0], device=self._psf.device))
 
-    @abc.abstractmethod
     def batch_call(self, batch):
         """
         Method for performing iterative reconstruction on a batch of images.
-        This implementation simply calls `apply` on each image in the batch.
-        Training algorithms are expected to override this method with a properly vectorized implementation.
+        This implementation is a properly vectorized implementation of FISTA.
 
         Parameters
         ----------
-        batch : :py:class:`~torch.Tensor` of shape (batch, depth, height, width, channels)
+        batch : :py:class:`~torch.Tensor` of shape (N, D, C, H, W)
```

Comment:
thanks for fixing order, but I would keep original terms (batch, depth, channels height, width, channels). Depth may not be obvious